### PR TITLE
Refactor BoardView to ViewModel & Fragment

### PR DESCRIPTION
**Pull request by raphaelseher**

### CODE REVIEW

**Comment by raphaelseher** [comment]
|File|Coverage [39.56%]|:green_apple:|
|:-|:-:|:-:|
|[CardDecks.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardDecks.java)|93.22%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|88.07%|:green_apple:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|86.24%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|83.5%|:green_apple:|
|[BoardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewModel.java)|0%|:green_apple:|
|[BoardViewData.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewData.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|
|[CardDetails.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FCardDetails.java)|0%|:green_apple:|
|[DetailedCardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardFragment.java)|0%|:green_apple:|
|[CardRowHelper.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FCardRowHelper.java)|0%|:green_apple:|
|[DebugHelper.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FDebugHelper.java)|0%|:green_apple:|
|[SingleEvent.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FSingleEvent.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|
|[MainActivity.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainActivity.java)|0%|:green_apple:|
|[CardView.java](https://github.com/SE2-Gwent/Gwent/blob/eaaa7f7533d532b7b1bab23d3e2bb217576c9441/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fcommon%2FCardView.java)|0%|:green_apple:|

|Total Project Coverage|29.21%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [39.54%]|:green_apple:|
|:-|:-:|:-:|
|[CardDecks.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardDecks.java)|93.22%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|88.07%|:green_apple:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|86.24%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|83.43%|:green_apple:|
|[BoardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewModel.java)|0%|:green_apple:|
|[BoardViewData.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewData.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|
|[CardDetails.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FCardDetails.java)|0%|:green_apple:|
|[DetailedCardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardFragment.java)|0%|:green_apple:|
|[CardRowHelper.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FCardRowHelper.java)|0%|:green_apple:|
|[DebugHelper.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FDebugHelper.java)|0%|:green_apple:|
|[SingleEvent.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FSingleEvent.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|
|[MainActivity.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainActivity.java)|0%|:green_apple:|
|[CardView.java](https://github.com/SE2-Gwent/Gwent/blob/9bfef0c4bdce5424598d9f8938c38dc6fd2ef332/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fcommon%2FCardView.java)|0%|:green_apple:|

|Total Project Coverage|29.2%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [37.09%]|:green_apple:|
|:-|:-:|:-:|
|[CardDecks.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardDecks.java)|93.22%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|88.07%|:green_apple:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|86.24%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|83.43%|:green_apple:|
|[FirebaseConnector.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FFirebaseConnector.java)|1.48%|:green_apple:|
|[BoardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewModel.java)|0%|:green_apple:|
|[BoardViewData.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewData.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|
|[CardDetails.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FCardDetails.java)|0%|:green_apple:|
|[DetailedCardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardFragment.java)|0%|:green_apple:|
|[CardRowHelper.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FCardRowHelper.java)|0%|:green_apple:|
|[DebugHelper.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FDebugHelper.java)|0%|:green_apple:|
|[SingleEvent.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FSingleEvent.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|
|[MainActivity.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainActivity.java)|0%|:green_apple:|
|[CardView.java](https://github.com/SE2-Gwent/Gwent/blob/d8c33f3be187c71c674ce239a8b0daf4eb0613ff/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fcommon%2FCardView.java)|0%|:green_apple:|

|Total Project Coverage|29.09%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [35.66%]|:green_apple:|
|:-|:-:|:-:|
|[CardDecks.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardDecks.java)|93.22%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|83.87%|:green_apple:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|83.67%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|47.71%|:green_apple:|
|[FirebaseConnector.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FFirebaseConnector.java)|1.53%|:green_apple:|
|[BoardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewModel.java)|0%|:green_apple:|
|[BoardViewData.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewData.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|
|[CardDetails.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FCardDetails.java)|0%|:green_apple:|
|[DetailedCardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardFragment.java)|0%|:green_apple:|
|[CardRowHelper.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FCardRowHelper.java)|0%|:green_apple:|
|[DebugHelper.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FDebugHelper.java)|0%|:green_apple:|
|[SingleEvent.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FSingleEvent.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|
|[MainActivity.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainActivity.java)|0%|:green_apple:|
|[CardView.java](https://github.com/SE2-Gwent/Gwent/blob/6fc31c726154a9129d43322e06f2094e45b0370d/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fcommon%2FCardView.java)|0%|:green_apple:|

|Total Project Coverage|28.18%|:green_apple:|
|:-|:-:|:-:|

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Sikredo) [inline .idea/misc.xml:1]: Lets exclude this file in the gitIgnore as it is in the .idea folder

File: app/src/main/java/at/aau/se2/gwent/MainActivity.java [hunk 2]
```diff
@@ -50,6 +50,10 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         Navigation.findNavController(this, R.id.nav_host_fragment_content_main)
             .navigate(R.id.detailed_card);
         return true;
+      case R.id.action_debug:
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/MainActivity.java:53]: Should be placed on the StartFragment

File: gamelogic/src/main/java/at/aau/se2/gamelogic/GameLogic.java [hunk 3]
```diff
@@ -242,14 +244,14 @@ private void initializeGame(Player currentPlayer, CardDecks cardDecks, ArrayList
 
   private void drawCards() {
     ArrayList<Card> cards = new ArrayList<Card>(gameField.getCardDeck(whoAmI).values());
-    if (cards.size() < 10) {
+    if (cards.size() < HAND_CARD_NUMBER) {
       Log.w(TAG, "CardDecks not setup");
```
Comment (Sikredo) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/GameLogic.java:248]: Should we do something in that case as there would be no cards?

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameFieldRows.java [hunk 4]
```diff
@@ -1,21 +1,16 @@
...
 
 public class GameFieldRows {
-  private ArrayList<Card> p1MeleeRow = new ArrayList<>();
-  private ArrayList<Card> p1RangeRow = new ArrayList<>();
-  private ArrayList<Card> p2MeleeRow = new ArrayList<>();
-  private ArrayList<Card> p2RangeRow = new ArrayList<>();
-
-  @Keep
-  public GameFieldRows() {}
+  private HashMap<String, Card> p1MeleeRow = new HashMap<>(10);
```
Comment (Sikredo) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/GameFieldRows.java:6]: Lets use here a final const for the fixed value of 10

**Review by kuttnig — APPROVED** [review]

File: app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+      if (card == null) continue;
+
+      CardView cardView = new CardView(getContext(), null);
+      // TODO: Replace drawable with cards drawable
+      cardView.setupWithCard(
+          card.getFirebaseId(), card.getPower(), card.getName(), R.drawable.an_craite_amorsmith);
+      rowLayout.getRoot().removeViewAt(i);
+      rowLayout.getRoot().addView(cardView, i);
+      cardView.setOnClickListener(
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java:217]: Lets move out the setOnClickListeners to a own private method as this is used here and in line 176-187

File: app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+
+      CardView cardView = new CardView(getContext(), null);
+      handLayout.getRoot().addView(cardView);
+      // TODO: Replace drawable with cards drawable
+      if (isOpponent) {
+        cardView.showAsPlaceholder(false);
+      } else {
+        cardView.setupWithCard(
+            entry.getKey(), card.getPower(), card.getName(), R.drawable.an_craite_amorsmith);
+        cardView.setOnClickListener(
```
Comment (Sikredo) [inline app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java:178]: Same here, lets move this out to a method

**Review by Sikredo — CHANGES_REQUESTED** [review]

File: app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java (hunk 5 above)
Comment (raphaelseher) [reply app/src/main/java/at/aau/se2/gwent/views/board/BoardFragment.java:217]: Sind zwei verschiedene methoden die gecallt werden. Da kriegen wir nit viel raus.
`didClickRowCard` und `didClickHandCard`

**Comment by raphaelseher** [comment]
|File|Coverage [33.53%]|:green_apple:|
|:-|:-:|:-:|
|[CardDecks.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCardDecks.java)|93.22%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|83.81%|:green_apple:|
|[GameField.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameField.java)|83.67%|:green_apple:|
|[GameFieldRows.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FGameFieldRows.java)|47.71%|:green_apple:|
|[FirebaseConnector.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FFirebaseConnector.java)|1.53%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2FstartMenu%2FStartFragment.java)|0%|:green_apple:|
|[SettingsFragment.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fsettings%2FSettingsFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdebug%2FGameDebugViewModel.java)|0%|:green_apple:|
|[GameDebugFragment.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdebug%2FGameDebugFragment.java)|0%|:green_apple:|
|[BoardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewModel.java)|0%|:green_apple:|
|[BoardViewData.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardViewData.java)|0%|:green_apple:|
|[BoardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fboard%2FBoardFragment.java)|0%|:green_apple:|
|[CardDetails.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FCardDetails.java)|0%|:green_apple:|
|[DetailedCardFragment.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardFragment.java)|0%|:green_apple:|
|[CardRowHelper.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FCardRowHelper.java)|0%|:green_apple:|
|[DebugHelper.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Futil%2FDebugHelper.java)|0%|:green_apple:|
|[SingleEvent.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fcomunication%2FSingleEvent.java)|0%|:green_apple:|
|[CardView.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fcommon%2FCardView.java)|0%|:green_apple:|
|[RulesFragment.java](https://github.com/SE2-Gwent/Gwent/blob/71e73e2d424b6e42feb3541a6ff5dba89241b448/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Frules%2FRulesFragment.java)|0%|:green_apple:|

|Total Project Coverage|27.36%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**